### DISCUSSION TITLE
Ability to select a default CDN for live streams (advanced feature)

### DISCUSSION THREAD

**Discussion by GermanCoding** [f1multiviewer/issue-tracker#379] [Feature requests] [answered]
Hello,

ever since I bought F1 TV (Pro) in 2020, I've had sporadic issues with it. On some race days it's fine, sometimes it buffers horribly throughout the session and is basically unusable.

Starting with this season, I'm using MultiViewer for F1 (it's awesome by the way!), because it enhances F1 TV so much. However, on live races I'm still having occasional issues with videos buffering all the time.

Today's race in Baku was particularly bad, with **all** streams (International stream, F1 live, OBC) completely breaking apart. The buffer health dropped to 0 seconds on all of them and never recovered. Setting a resolution > 360p on *any* stream caused the buffer health to consistently drop to 0, even with target latency set to > 60 seconds. I'm using a 250 Mbit internet connections, so bandwidth should not be a problem.

Playing around with quality and debug settings, I noticed a pattern: The videos (F1 Live + OBC) only struggled when the `cdnType` was `cloudfront` (i.e. Amazon's CDN). When the `cdnType` was `akamai` I had 0 issues and could open dozens of International streams simultaneously without issue.

In order to get the CDN to be set to akamai, I pressed right click -> Reload until the CDN was set to akamai. This is a bit annoying when switching between OBCs, because sometimes I have to reload them several times to get Akamai.

Retroactively, I believe the issues I've had for 3 years might have always been a CDN issue. I didn't know it back then, but perhaps the days when everything was fine where the days where I coincidentally was routed to Akamai CDN.

I don't know why my ISP (German Telekom) seems to struggle with Cloudfront, it appears to be only F1 TV that it struggles with. Anyway, since Akamai works flawlessly for me, I was wondering whether MultiViewer could have an advanced option that automatically prefers a given CDN?

**Comment by f1multiviewer** [comment, marked as answer]
MultiViewer had this, at some point, but I removed this as it may cause issues. F1TV uses a weighted round-robin load balancing algorithm and MultiViewer would interfere with this. I don't feel super comfortable doing this, also because the only way to get the other CDN is to try and request another playback URL until you get the right CDN, which doesn't seem right.
